app/models/ama.rb

```diff
@@ -46,12 +46,12 @@ def send_ama_notifciation
   end
 
   def open?
-    now_time = Time.now
-    start_date <= now_time && end_date >= now_time
+    (start_date..end_date).cover?(Time.now)
   end
 
   before_validation do
-    self.end_date = start_date + 1.hour
+    self.end_date = start_date + 1.hour if start_date.to_i >= end_date.to_i
+    self.end_date ||= start_date + 1.hour
```

Comment:
It might be best to tweak this to one conditional, so like

```ruby
if end_date.blank? || start_date >= end_date
  self.end_date = start_date + 1.hour
end
```

Also you shouldn't need those `to_i` calls — Ruby has fantastic operator overloading, so you should be able to compare Dates directly